ToolBox/LeanDataWriter.cs

```diff
@@ -45,13 +46,15 @@ public class LeanDataWriter
         /// <param name="dataDirectory">Base data directory</param>
         /// <param name="resolution">Resolution of the desired output data</param>
         /// <param name="tickType">The tick type</param>
-        public LeanDataWriter(Resolution resolution, Symbol symbol, string dataDirectory, TickType tickType = TickType.Trade)
+        /// <param name="appendToZips">Append to existing zip rather than replace them.  Entries will still be replaced entirely.</param>
+        public LeanDataWriter(Resolution resolution, Symbol symbol, string dataDirectory, TickType tickType = TickType.Trade, bool appendToZips = false)
```

Comment:
The new argument `appendToZips` will be used only for Futures, right?

If so, I suggest to remove it as argument and define its value based on `_securityType`.

Reply:
@Jay-Jay-D - Yes, I wondered about that too.  My thinking is that I wanted to allow the overwriting functionality to still be possible, regardless of the security type.  But it could still be valuable to default to that for futures.

I've made a change that allows that value to be defaulted to the security type provided.  That way it would be possible to create Lean archives without appending if so desired.

I'm happy to get rid of the optional input entirely and base it solely on security type if you think that's better.

Reply:
Yes, please remove the optional argument, it'll make it less error-prone.

Reply:
@Jay-Jay-D - I've removed the optional argument in favour of deriving `appendToZips` from the trade type.

I also called into a common constructor so that `appendToZips` logic didn't exist in 2 places.